modules/kit-elements-defaults/utils/settings-sanitizer.php

```diff
@@ -95,16 +95,25 @@ public function remove_invalid_settings() {
 
 		$controls = $this->pending_element->get_controls();
 
-		$this->pending_settings = ( new Collection( $this->pending_settings ) )
-			->only( array_merge( array_keys( $controls ), static::SPECIAL_SETTINGS ) )
-			->map( function ( $value, $key ) use ( $controls ) {
-				if ( ! in_array( $key, static::SPECIAL_SETTINGS, true ) ) {
-					return $value;
-				}
+		if ( ! $controls ) {
+			return $this;
+		}
+
+		$available_settings = $this->get_avaliable_settings( $controls );
 
-				return array_intersect_key( $value, $controls );
-			} )
-			->all();
+		foreach ( $this->pending_settings as $key => $value ) {
+			if ( in_array( $key, static::SPECIAL_SETTINGS, true ) ) {
+				// When the key is special settings like __dynamic__ or __globals__ it should run over the values.
```

Comment:
Not sure that this comment is needed, consider adding a variable for the condition above:
$is_special_setting - like you had in the first place.

Reply:
Done